lib/gems/pending/util/require_with_logging.rb

```diff
@@ -30,16 +31,28 @@ def log_require(path, mode, timing = nil)
     $req_depth += 1 if mode == :enter || mode == :reenter
   end
 
+  def log_require_time
+    Time.now
+  end
+
+  def log_require_allocations
+    GC.stat[:total_allocated_objects]
+  end
+
+  def log_require_mb
+    Sys::ProcTable.ps(Process.pid).smaps.pss / 1_000_000.0
```

Comment:
Note, this is linux only.  😢

Reply:
The `Sys::ProcTable.ps` does a [directory scan](https://github.com/djberg96/sys-proctable/blob/master/lib/linux/sys/proctable.rb#L114) that I really don't like when getting the info for a single process, so it is pretty memory hungry for what we are trying to do (just getting mem value of the process).  I have simplified slimed that down as much as I could using the existing functions of `sys-proctable` in a extra data initializer I was using for a `MiqServer` process:

https://gist.github.com/NickLaMuro/2d494d9fc0050f4e705ebab58437d108#file-miq_server_mem_log_patch-local-rb-L20

But it probably would be better to consider pushing a common interface into `sys-proctable` for getting simple things like memory usage and memory info.  Something that has been on my `WANTTODO` list, but haven't had the cycles to go and do it.  #firstWorldMemoryLeakProblems

Reply:
I also have a complex version of this that uses whats available in `manageiq-performance` (in a overly long WIP PR):

https://github.com/ManageIQ/manageiq-performance/pull/27/commits/0d9a3ee

Lifts some parts of that from the `get_process_mem` gem, and uses sys-proctable as a fall back (did not know about the faster version I came up with above when I wrote that, FYI).  Since `get_process_mem` portion is reading from the `/proc/pid/status` file is most undoubtedly faster, since it is much shorter.

The other thing I am curious about is if for something like this, where we are reading from the same file A BUNCH of times (well, file-like-interface), if maintaining a single keeping a File object memoize, and opening, closing, and rewinding as needed would be more performant.  Would something interesting to try.